drivers/iio/adc/ad400x.c

```
@@ -575,6 +608,28 @@ static int ad400x_probe(struct spi_device *spi)
 	if (ret < 0)
 		return ret;
 
+	/* Hardware gain only applies to ADAQ devices */
+	st->gain_milli = 1000;
+	if (device_property_present(&spi->dev, "adi,gain-milli")) {
```

nit: valid checkpatch complain in here. Please fix :)

---

Also, I'm assuming we should have some constrains for the accepted values? Take that into consideration and don't assume people will always do the right thing in DT/FW (always better to fail early :)). Also no point in doing a first `device_property_present()` call.

---

Yes, we had this discussion on the mailing list. I scraped that old scale implementation that was in this PR and replaced it by the scale support for AD4000 series that was discussed and improved in the upstream process. I think it solves all the issues related to the old implementation.